src/app/object-schema/object-schema.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ObjectSchemaFormControl } from '../formControls/ObjectSchemaFormControl';
+import { ObjectSchemaEditorService } from '../services/object-schema-editor/object-schema-editor.service';
+
+@Component({
+  selector: 'object-schema',
+  templateUrl: './object-schema.component.html',
+  styleUrls: ['./object-schema.component.scss']
+})
+export class ObjectSchemaComponent implements OnInit {
+
+  private maxLevel: number = 5; 
+  @Input() public field!: ObjectSchemaFormControl;
+  public schemaObjectEditorService: ObjectSchemaEditorService; 
+  public panelOpenState = true;
+  public schemaTypes : string[] = [
```

Comment:
These are already supplied by the `EditorService`. You can either move it to the `ObjectSchemaEditorService` but would require updating all the components as well. The other alternative would be to also inject the `EditorService` as another dependency. One thing I would like to avoid if possible is having a Service -> Service dependency hierarchy. I was already forced to do this with the `ValidationService` but should be temporary once I figure out a cleaner pattern.

Reply:
Ok, I can grab this from the editor service itself and inject it. Thats easy.